### PR TITLE
feat: Add Icon Picker font awesome component - EXO-65005_66218 - Meeds-io/MIPs#72

### PR DESCRIPTION
**Pull request by IlhemEssaadi**

### CODE REVIEW

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationNodeDrawer.vue [hunk 1]
```diff
@@ -100,13 +100,15 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
               class="d-flex flex-grow-1 full-width">
               <div class="me-2 ms-auto">
                 <v-btn
+                  v-if="isDefaultIcon"
                   id="deleteIconButton"
                   :title="$t('siteNavigation.btn.deleteIcon.title')"
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationNodeDrawer.vue]: we should have the title displayed like for visibility info case, same for changeIcon case

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigation.vue [hunk 2]
```diff
@@ -22,6 +22,7 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
     <site-navigation-node-drawer v-if="siteNavigationDrawerOpened" />
     <site-navigation-element-drawer v-if="siteNavigationDrawerOpened" />
     <layout-notification-alerts v-if="siteNavigationDrawerOpened" />
+    <font-awesome-picker-drawer />
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigation.vue]:
```suggestion
    <font-awesome-picker-drawer v-if="siteNavigationDrawerOpened" />
```

File: layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties [hunk 3]
```diff
@@ -28,6 +28,8 @@ siteNavigation.label.icon.title=Icon of the node
 siteNavigation.label.icon.description=The image is displayed with the node label in the left navigation
 siteNavigation.btn.deleteIcon.title=Delete Icon
 siteNavigation.btn.changeIcon.title=Change Icon
+siteNavigation.label.selectIcon.title=Select Icon
```
Comment (azayati) [inline layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties]:
```suggestion
siteNavigation.fontAwesomePickerDrawer.title=Select Icon
```

File: layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties [hunk 4]
```diff
@@ -28,6 +28,8 @@ siteNavigation.label.icon.title=Icon of the node
...
+siteNavigation.label.btn.showMore=show More
```
Comment (azayati) [inline layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties]:
```suggestion
siteNavigation.btn.showMore.title=Show More
```

File: layout-management-webapps/package.json [hunk 5]
```diff
@@ -27,5 +27,9 @@
     "webpack": "^5.72.0",
     "webpack-cli": "^4.9.2",
     "webpack-merge": "^5.8.0"
+  },
+  "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.14",
+    "@fortawesome/free-solid-svg-icons": "^5.7.1"
```
Comment (azayati) [inline layout-management-webapps/package.json]:
```suggestion
    "@fortawesome/fontawesome-svg-core": "^1.2.36",
    "@fortawesome/free-solid-svg-icons": "^5.15.4"
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue [hunk 6]
```diff
@@ -0,0 +1,196 @@
+<!--
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue]: Rename to NodeIconPickerDrawer.vue

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+<template>
+  <exo-drawer
+    ref="fontAwesomePickerDrawer"
+    id="fontAwesomePickerDrawer"
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue]:
```suggestion
    ref="nodeIconPickerDrawer"
    id="nodeIconPickerDrawer"
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+          fas fa-arrow-left
+        </v-icon>
+        <span class="ms-2"> {{ $t('siteNavigation.label.selectIcon.title') }}</span>
+      </div>
+    </template>
+    <template slot="content">
+      <v-card-text class="d-flex pb-2">
+        <v-text-field
+          v-model="keyword"
+          placeholder="filter by name"
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue]: add an internationalized label

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+        <v-container id="iconPicker">
+          <v-row class="ma-0">
+            <v-col 
+              v-for="icon in icons"
+              :key="icon.iconValue"
+              cols="3"
+              class="pa-0 pa-1">
+              <v-sheet 
+                class="rounded-lg clickable "
+                :class="selectedIcon === icon ? 'grey-background' : 'light-grey-background'"
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue]: computed

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+        </v-btn>
+      </div>
+    </template>
+  </exo-drawer>
+</template>
+
+
+<script>
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { fas } from '@fortawesome/free-solid-svg-icons';
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue]: no imports allowed directly in vuejs files:
https://github.com/Meeds-io/meeds/blob/develop/CODING_GUIDELINES.md#devstd-29

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+      } else {
+        return Array.from(this.allIcons).slice(0, this.iconsNumber);
+      }
+    },
+    extraClass() {
+      return this.selectedIcon ? 'selectedIcon' : 'light-grey-background';
+    }
+  },
+  created() {
+    this.$root.$on('open-font-awesome-picker-drawer', this.open);
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/FontAwesomePickerDrawer.vue]:
```suggestion
    this.$root.$on('open-node-icon-picker-drawer', this.open);
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationNodeDrawer.vue [hunk 12]
```diff
@@ -317,6 +324,9 @@ export default {
       this.open(navigationNode);
     });
     this.$root.$on('save-node-with-page', this.saveNode);
+    this.$root.$on('update-icon-node', (icon) => {
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationNodeDrawer.vue]:
```suggestion
    this.$root.$on('update-node-icon', (icon) => {
```

**Review by azayati — CHANGES_REQUESTED** [review]

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/NodeIconPickerDrawer.vue [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+
+
+<script>
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+
+// Add all solid Font Awesome icons to the library
+library.add(fas);
+export default {
+  data () {
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/NodeIconPickerDrawer.vue]:
```suggestion
  data() {
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/NodeIconPickerDrawer.vue [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+        icon.iconColor = 'grey-background';
+        this.disabled = false;
+        this.selectedIcon = icon;
+      }
+    },
+    saveIcon() {
+      this.$root.$emit('update-node-icon', this.selectedIcon.iconValue);
+      this.$nextTick().then(() => this.close());
+    },
+    filterIcons (event) {
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/NodeIconPickerDrawer.vue]:
```suggestion
    filterIcons(event) {
```

**Review by azayati — CHANGES_REQUESTED** [review]
- https://github.com/exoplatform/layout-management/pull/237/files#r1328854141

File: layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties [hunk 15]
```diff
@@ -28,6 +28,9 @@ siteNavigation.label.icon.title=Icon of the node
 siteNavigation.label.icon.description=The image is displayed with the node label in the left navigation
 siteNavigation.btn.deleteIcon.title=Delete Icon
 siteNavigation.btn.changeIcon.title=Change Icon
+siteNavigation.fontAwesomePickerDrawer.title=Select Icon
```
Comment (azayati) [inline layout-management-webapps/src/main/resources/locale/portlet/layoutManagement/SiteNavigationPortlet_en.properties]:
```suggestion
siteNavigation.nodeIconPickerDrawer.title=Select Icon
```

File: layout-management-webapps/src/main/webapp/skin/less/siteNavigation.less [hunk 16]
```diff
@@ -13,6 +13,12 @@
   }
 }
 
+#fontAwesomePickerDrawer {
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/skin/less/siteNavigation.less]:
```suggestion
#nodeIconPickerDrawer {
```

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationNodeDrawer.vue [hunk 17]
```diff
@@ -116,7 +118,8 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
                   icon
                   outlined
                   dark
-                  small>
+                  small
+                  @click="openFontAwesomePicker">
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationNodeDrawer.vue]:
```suggestion
                  @click="openNodeIconPickerDrawer">
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/NodeIconPickerDrawer.vue [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+        return Array.from(this.allIcons).slice(0, this.iconsNumber);
+      }
+    },
+  },
+  created() {
+    this.$root.$on('open-node-icon-picker-drawer', this.open);
+  },
+  methods: {
+    open() {
+      this.allIcons =this.fontLibrary.icons.map(icon => ({
```
Comment (azayati) [inline layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/NodeIconPickerDrawer.vue]:
```suggestion
      this.allIcons = this.fontLibrary.icons.map(icon => ({
```

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**